bots/pr/src/main/java/org/openjdk/skara/bots/pr/ApprovalCommand.java

```diff
@@ -170,4 +172,15 @@ public boolean multiLine() {
     private void showHelp(PrintWriter reply) {
         reply.println("usage: `/approval [<id>] (request|cancel) [<text>]`");
     }
+
+    public static String realTargetRef(PullRequest pr) {
+        var targetRef = pr.targetRef();
+        var matcher = PRE_INTEGRATE_BRANCH_PATTERN.matcher(targetRef);
+        if (!matcher.matches()) {
+            return targetRef;
+        }
+        String id = matcher.group(1);
+        var dependentPR = pr.repository().pullRequest(id);
+        return realTargetRef(dependentPR);
+    }
```

Comment:
There is a class `org.openjdk.skara.forge.PreIntegrations` that handles these things. I think this method should be added there (and be integrated with existing methods).